ex2_hal/athena/equipment_handler/source/ina209.c

```diff
@@ -0,0 +1,247 @@
+/*
+ *  INA209.c
+ *  Drivers for INA209 Current Sense
+ *  See Datasheet for possible register values
+ *  https://www.ti.com/lit/ds/symlink/ina209.pdf
+ *
+ *  INA209 registers are Most Significant Byte first - LSB PEEPS BEWARE
+ *  (MSB peeps just toss the _flip_byte_order() function)
+ *  Created on: May 10, 2022
+ *  Author: Liam Droog
+ */
+
+#include <ina209.h>
+#include <i2c_io.h>
+#include <FreeRTOS.h>
+#include <os_task.h>
+#include "printf.h"
+
+uint16_t CALI_REG = 0xDA73;
+uint16_t POWER_OLREG = 0x0100;
+uint16_t ZEROREG = 0x0000;
+uint16_t CONTROL_REG = 0x0082;
+
+int ina209_Write1ByteReg(uint8_t addr, uint8_t reg_addr, uint8_t data) {
+    uint8_t buf[2];
+    buf[0] = reg_addr;
+    buf[1] = data;
+    return i2c_Send(i2cREG2, addr, 2, &buf);
+}
+
+int ina209_Write2ByteReg(uint8_t addr, uint8_t reg_addr, uint16_t data) {
+    uint8_t buf[3];
+    buf[0] = reg_addr;
+    buf[1] = (data & 0xFF00) >> 8; // 00FF
+    buf[2] = data & 0xFF;          // FF00
+    return i2c_Send(i2cREG2, addr, 3, &buf);
+}
+
+int ina209_Read1ByteReg(uint8_t addr, uint8_t reg_addr, uint8_t *val) {
+    // TODO: make this use error code return instead
+    uint8_t value = 0;
+
+    i2c_Send(i2cREG2, addr, 1, &reg_addr);
+
+    i2c_Receive(i2cREG2, addr, 1, &value);
+
+    return value;
+}
+
+int ina209_Read2ByteReg(uint8_t addr, uint8_t reg_addr, uint16_t *val) {
+    // TODO: make this use error code return instead
+    uint8_t data[2] = {0};
+    uint16_t value = 0;
+
+    if (i2c_Send(i2cREG2, addr, 1, &reg_addr) == -1) {
+        return -1;
+    }
+
+    if (i2c_Receive(i2cREG2, addr, 2, &data) == -1) {
+        return -1;
+    }
+    *val = (((uint16_t)(data[0])) << 8) | data[1];
+
+    return 0;
+}
+
+void ina209_get_configuration(uint8_t addr, uint16_t *retval) {
```

Comment:
None of these functions do anything with the return values of the register read functions...